src/main/java/au/org/ala/biocache/service/AuthService.java

```diff
@@ -184,26 +193,111 @@ public void run() {
             logger.info("authCache requesting: " + jsonUri);
-            roles = (List) restTemplate.postForObject(jsonUri, null, Map.class).get("roles");
+            roles.addAll((List) restTemplate.postForObject(jsonUri, null, Map.class).getOrDefault("roles", Collections.EMPTY_LIST));
         }
         return roles;
     }
 
+    @Deprecated
     public Map<String,?> getUserDetails(String userId) {
         Map<String,?> userDetails = new HashMap<>();
-        if(StringUtils.isNotBlank(userDetailsUrl)){
+        if (StringUtils.isNotBlank(userDetailsUrl)){
             final String jsonUri = userDetailsUrl + userDetailsPath + "?userName=" + userId;
             logger.info("authCache requesting: " + jsonUri);
             userDetails = (Map) restTemplate.postForObject(jsonUri, null, Map.class);
         }
         return userDetails;
     }
+
+    /**
+     * Authentication for download users has 3 routes:
+     *
+     * 1) Check for JWT / OAuth- user is retrieved from UserPrincipal along with a set of roles, supplied email address is ignored...
+     * 2) Legacy API Key and X-Auth-Id - email address retrieved from CAS/Userdetails - email address is ignored...
+     * 3) Email address supplied (Galah) - email address is verified - no sensitive access
+     */
+    public Optional<AlaUser> getDownloadUser(DownloadRequestDTO downloadRequestDTO, HttpServletRequest request) {
+
+        // 2) Check for JWT / OAuth
+        if (request.getUserPrincipal() != null && request.getUserPrincipal() instanceof PreAuthenticatedAuthenticationToken){
+            return Optional.of((AlaUser) ((PreAuthenticatedAuthenticationToken) request.getUserPrincipal()).getPrincipal());
+        }
+
+        // 3) Email address supplied (Galah / ala4r) - email address is verified - no roles, no sensitive access
+        if (emailOnlyEnabled && downloadRequestDTO.getEmail() != null) {
+            try {
+                new InternetAddress(downloadRequestDTO.getEmail()).validate();
+                // verify the email address is registered
+                return lookupAuthUser(downloadRequestDTO.getEmail(), false);
+            } catch (AddressException e) {
+                // invalid email
+                logger.info("Email only download request failed - invalid email " + downloadRequestDTO.getEmail());
+            }
+        }
+
+        return Optional.empty();
+    }
+
+
+    /**
+     * Authentication for download users has 3 routes:
+     *
+     * 1) Check for JWT / OAuth- user is retrieved from UserPrincipal along with a set of roles, supplied email address is ignored...
+     * 2) Legacy API Key and X-Auth-Id - email address retrieved from CAS/Userdetails - email address is ignored...
+     */
+    public Optional<AlaUser> getRecordViewUser(HttpServletRequest request) {
+        // 2) Check for JWT / OAuth
+        if (request.getUserPrincipal() != null && request.getUserPrincipal() instanceof PreAuthenticatedAuthenticationToken){
```

Comment:
It appears when `request.getUserPrincipal()` is not null, it is not of the type `PreAuthenticatedAuthenticationToken`. 
I tested authenticated user requests from ala-hub  and this never resolves `true` when request.getUserPrincipal() is valid.

Reply:
Any ideas @djtfmartin , @brucehyslop ?